graspologic/cluster/autogmm.py

```diff
@@ -44,7 +46,7 @@ class AutoGMMCluster(BaseCluster):
         If label_init is given, min_components must match number of unique labels
         in label_init.
 
-    affinity : {'euclidean','manhattan','cosine','none', 'all' (default)}, optional
+    affinity : {'euclidean','manhattan','cosine','none', 'all' (default)}
```

Comment:
The draft PR description covers why these aren't really Optional - they have a default, but it can't be None - it has to be either str, List[str], or np.ndarray (with a value type of str)

Reply:
this isn't my code, but personally i always took "optional" in python meaning it is a kwarg, and one doesn't need to specify it for the code to run, not that it is a kwarg that can be None and default is None.

Reply:
For sure - it's not unreasonable to say that a "parameter not provided will get a default and thus is conceptually an optional value".  The problem is that https://docs.python.org/3.7/library/typing.html#typing.Optional means something very different, and using the same phrasing while using typing is just going to cause a lot of confusion later.

Better to just call it a named parameter that has a default - then you don't run into any confusion later

Reply:
to add to what i've said above, [as i've said previously: conventions can always be broken, but] here i believe it is also a very wide-spread convention to mean explicitly that:

> "parameter not provided will get a default and thus is conceptually an optional value"

so we will be confusing a lot of people with an overloaded definition of optional, even though Optional is a part of python library.

example: [click](https://numpy.org/doc/stable/reference/random/generated/numpy.random.randint.html?highlight=randint#numpy.random.randint)

Reply:
i also would be confused by the above for what it's worth, maybe worth chatting about

Reply:
or i should clarify - the new line does not confuse me at all, the word default makes it clear to me, but this is defs not how we've been doing things or I have seen them done in many packages. so not opposed to this practice but it comes as a surprise to me